source/server/http_test_server_filter.cc

```diff
@@ -79,6 +104,10 @@ HttpTestServerDecoderFilter::decodeTrailers(Envoy::Http::RequestTrailerMap&) {
 void HttpTestServerDecoderFilter::setDecoderFilterCallbacks(
     Envoy::Http::StreamDecoderFilterCallbacks& callbacks) {
   decoder_callbacks_ = &callbacks;
+  // TODO(#486): this adds locking in the hot path. Consider moving this into a separate
+  // extension, which will also allow tracking multiple points via configuration.
+  last_request_delta_ns_ =
```

Comment:
Did we (or can we) verify the performance implication of this? How likely is this going to cause significant regressions in performance tests that use the test server?

If we can't verify it, we may be better off keeping this behind a feature flag to begin with? Or is it already on the path only when we ask for it by setting `emit_previous_request_delta_in_response_header`?

Reply:
I couldn't measure a significant difference in manual tests. Having said that, I'd love to move this into its own extension because intuitively I'd say that there's no way that this can have a positive impact.
The locking now is unconditional, and it is hard to make it conditional because the configuration for it can be changed on a per-request basis. 

But a separate extension would allow one to easily take this out of the equation altogether, and possibly also allow one to inject it into multiple places in the server side extension pipeline if desired so, emitting these timings from various places. 
If you don't mind me bloating this PR a bit, I'd be up to move this into a new extension.

Reply:
Considering the unknown performance impact, I think it would be worthwhile to move this to a new extension in this PR. Thank you for looking into it.